File: backend/models/category.py
"""Category model (hierarchical)."""
from __future__ import annotations
from datetime import datetime, timezone
from pydantic import Field
from .base import BaseDocument, PyObjectId


class Category(BaseDocument):
    name: str
    slug: str
    icon_url: str | None = None  # emoji or URL
    parent_id: PyObjectId | None = None
    sort_order: int = 0
    is_active: bool = True
    is_deleted: bool = False
    created_at: str = Field(default_factory=lambda: datetime.now(timezone.utc).isoformat())
    updated_at: str = Field(default_factory=lambda: datetime.now(timezone.utc).isoformat())

File: backend/src/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema({
  name: { type: String, required: true },
  slug: { type: String, required: true, unique: true },
  icon_url: { type: String, default: null },
  image: { type: String, default: null },
  description: { type: String, default: null },
  category_type: { type: String, enum: ['product', 'service', null], default: null, index: true },
  parent_id: { type: String, default: null, index: true },
  sort_order: { type: Number, default: 0 },
  is_active: { type: Boolean, default: true },
  is_deleted: { type: Boolean, default: false },
}, {
  timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' },
});

categorySchema.index({ is_deleted: 1 });

module.exports = mongoose.model('Category', categorySchema, 'categories');

